File: solutions/kotlin/leap/1/src/main/kotlin/Leap.kt
data class Year(val year: Int) {

    /**
     * @return true if the given year is a leap year
     * @see <a href="https://en.wikipedia.org/wiki/Leap_year">Leap year</a>
     * Note: A leap year is a year divisible by 4, except for years divisible by 100 unless they are also divisible by 400.
     */

    val isLeap: Boolean = year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)
}

File: solutions/dart/leap/1/lib/leap.dart
class Leap {
  /**
   * Determines if the given year is a leap year.
   * A leap year is defined as:
   * - It is divisible by 4, but not divisible by 100, OR
   * - It is divisible by 400.
   */
  bool leapYear(int year) {
    bool divisibleBy4 = year % 4 == 0;
    bool divisibleBy100 = year % 100 == 0;
    bool divisibleBy400 = year % 400 == 0;

    return (divisibleBy4 && !divisibleBy100) || divisibleBy400;
  }
}
